db/attachment.go

```diff
@@ -146,11 +146,19 @@ func (db *Database) retrieveAncestorAttachments(doc *document, parentRev string,
 func (db *Database) loadBodyAttachments(body Body, minRevpos int) (Body, error) {
 
 	body = body.ImmutableAttachmentsCopy()
-	for _, value := range BodyAttachments(body) {
+	for attachmentName, value := range BodyAttachments(body) {
 		meta := value.(map[string]interface{})
 		revpos, ok := base.ToInt64(meta["revpos"])
 		if ok && revpos >= int64(minRevpos) {
-			key := AttachmentKey(meta["digest"].(string))
+			digest, ok := meta["digest"]
+			if !ok {
+				return nil, fmt.Errorf("Unable to load attachment with name: %v and revpos: %v due to missing digest field", attachmentName, revpos)
```

Comment:
This error message would be a lot more useful with the doc id included.  It doesn't look like the caller of loadBodyAttachments adds the doc id to the error - it just returns this error.  I think we should we either pass docid in to loadBodyAttachments, or have the caller wrap the error with something that includes the doc id?